### PR TITLE
Specify Cohere's model name

### PR DESCRIPTION
**Pull request by AmrMKayid**
Right now the cohere model `large` is hard coded. This PR enables specifying which cohere model to use for embeddings the documents and queries

### CODE REVIEW

**Review by jcudit — APPROVED** [review]
nice addition, thanks for making this more flexible!

File: embed/cli.py [hunk 1]
```diff
@@ -28,7 +28,12 @@ def cli(ctx):
 @click.option('--output-file', help='path to output file of embeddings')
 @click.option('--context-window', help='number of surrounding blocks included while embedding', default=6)
 @click.option('--api_token', help='Cohere API token', default=None)
-def blocks(input_blocks: str, output_file: str, context_window: int, api_token: typing.Optional[str] = None) -> None:
+@click.option('--model_name', help='Cohere Model name', default='large')
```
Comment (egrefen) [inline embed/cli.py]:
```suggestion
@click.option('--model_name', help='Cohere model name', default='large')
```

File: embed/cli.py [hunk 2]
```diff
@@ -64,7 +69,12 @@ def blocks(input_blocks: str, output_file: str, context_window: int, api_token:
 @click.option('--output-file', help='path to output file of embeddings')
 @click.option('--context-window', help='number of surrounding blocks included while embedding', default=6)
 @click.option('--api_token', help='Cohere API token', default=None)
-def localtxt(root_dir, output_file, context_window, api_token: typing.Optional[str] = None):
+@click.option('--model_name', help='Cohere Model name', default='large')
```
Comment (egrefen) [inline embed/cli.py]:
```suggestion
@click.option('--model_name', help='Cohere model name', default='large')
```

File: server/app.py [hunk 3]
```diff
@@ -34,7 +34,7 @@ class SearchResponse(BaseModel):
 def init_v2():
     global client
     log.info('fetching v2 block embeddings from local storage')
-    client = SearchClient('/data/embeddings.npz')
+    client = SearchClient('/data/embeddings.npz')  # TODO(amr): specify the model name here
```
Comment (egrefen) [inline server/app.py]: Resolve TODO please.

File: server/search/cli.py [hunk 4]
```diff
@@ -25,12 +25,13 @@ def cli(ctx):
 @click.option('--version', help='specify version of search algorithm and embedding file format', default=2)
 @click.option('--input-embeddings', help='path to input file of embeddings')
 @click.option('--num-results', default=3)
-def blocks(input_embeddings: str, num_results: int) -> None:
+@click.option('--model_name', help='Cohere Model name', default='large')
```
Comment (egrefen) [inline server/search/cli.py]:
```suggestion
@click.option('--model_name', help='Cohere model name', default='large')
```

**Review by egrefen — CHANGES_REQUESTED** [review]
Thanks for the PR. There is one todo that needs resolving. Beyond that, the suggested changes are superficial.

File: server/app.py (hunk 3 above)
Comment (AmrMKayid) [reply server/app.py]: resolved!

File: server/app.py [hunk 5]
```diff
@@ -34,7 +35,7 @@ class SearchResponse(BaseModel):
 def init_v2():
     global client
     log.info('fetching v2 block embeddings from local storage')
-    client = SearchClient('/data/embeddings.npz')
+    client = SearchClient('/data/embeddings.npz', model_name=os.environ.get('COHERE_MODEL_NAME', 'large'))
```
Comment (egrefen) [inline server/app.py]:
```suggestion
    model_name = os.environ.get('COHERE_MODEL_NAME', 'large')
    client = SearchClient('/data/embeddings.npz', model_name=model_name)
```

**Review by egrefen — APPROVED** [review]
Good to merge after the suggested change is made.

**Pull request merged**